src/reflect/scala/reflect/internal/tpe/TypeMaps.scala

```diff
@@ -663,9 +663,8 @@ private[internal] trait TypeMaps {
   }
 
   /** A base class to compute all substitutions */
-  abstract class SubstMap[T](from: List[Symbol], to: List[T]) extends TypeMap {
-    // OPT this check was 2-3% of some profiles, demoted to -Xdev
-    if (isDeveloper) assert(sameLength(from, to), "Unsound substitution from "+ from +" to "+ to)
```

Comment:
We could keep `from` as a field of this class rather than as an abstract method. That would avoid virtual method overhead for its callers.